srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+    .. code-block:: c
+
+        bool set_custom_type_dynamic_data(
+                DDS_DynamicData & data,
+                unsigned long key,
+                int targetDataLen)
+        {
+            DDS_ReturnCode_t retcode;
+            char test_string[SIZE_TEST_STRING]; //size of member_name
+            bool success = true;
+            DDS_DynamicData customTypeData(NULL, DDS_DYNAMIC_DATA_PROPERTY_DEFAULT);
+            DDS_DynamicData testSeqData(NULL, DDS_DYNAMIC_DATA_PROPERTY_DEFAULT);
+
+            retcode = data.bind_complex_member(
+                    customTypeData,
+                    "custom_type",
+                    DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr,
+                        "bind_complex_member(custom_type) failed: %d.\n",
+                        retcode);
+                success = false;
+            }
+
+            retcode = customTypeData.set_long(
+                    "test_long",
+                    DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                    key);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr, "set_long(test_long) failed: %d.\n", retcode);
+                success = false;
+            }
+
+            retcode = customTypeData.set_long(
+                    "test_enum",
+                    DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                    ENUM1);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr, "set_long(test_enum) failed: %d.\n", retcode);
+                success = false;
+            }
+
+            if (snprintf(test_string, SIZE_TEST_STRING, "Hello World! %lu", key) < 0) {
+                success = false;
+            }
+            retcode = customTypeData.set_string(
+                    "test_string.test_string",
+                    DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                    test_string);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr, "set_string(test_string) failed: %d.\n", retcode);
+                success = false;
+            }
+
+            retcode = customTypeData.bind_complex_member(
+                    testSeqData,
+                    "test_seq",
+                    DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr,
+                        "bind_complex_member(testSeqData) failed: %d.\n",
+                        retcode);
+                success = false;
+            }
+            retcode = testSeqData.set_long_seq(
+                        "test_seq",
+                        DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                        longSeq);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr, "set_long(test_seq) failed: %d.\n", retcode);
+                success = false;
+            }
+            retcode = customTypeData.unbind_complex_member(testSeqData);
```

Comment:
Is the unbinding required to finalize the setting of the data?

If so, would be useful to have a comment making that statement.  Not sure how familiar people will be with the proper usage of the DynamicData API (I myself am a neophyte in this respect).